app/questionnaire/questionnaire_manager.py

```diff
@@ -56,7 +56,7 @@ def validate_all_answers(self):
 
         return True, None
 
-    def update_questionnaire_store(self, location):
+    def update_questionnaire_store(self, location, save=True):
```

Comment:
Maybe a save method would be more explicit than a flag which modifies the behaviour